tests/lib/nested.sh

```diff
@@ -30,32 +34,52 @@ create_assertions_disk(){
     umount /mnt/assertions && rm -rf /mnt/assertions
 }
 
-create_nested_core_vm(){
-    # determine arch related vars
+get_qemu_for_nested_vm(){
     case "$NESTED_ARCH" in
     amd64)
-        QEMU="$(command -v qemu-system-x86_64)"
+        echo "$(command -v qemu-system-x86_64)"
         ;;
     i386)
-        QEMU="$(command -v qemu-system-i386)"
+        echo "$(command -v qemu-system-i386)"
         ;;
     *)
         echo "unsupported architecture"
         exit 1
         ;;
     esac
+}
+
+get_image_url_for_nested_vm(){
+    case "$NESTED_SYSTEM" in
+    xenial|trusty)
+        echo "https://cloud-images.ubuntu.com/${NESTED_SYSTEM}/current/${NESTED_SYSTEM}-server-cloudimg-${NESTED_ARCH}-disk1.img"
+        ;;
+    bionic)
+        echo "https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-${NESTED_ARCH}.img"
+        ;;
+    *)
+        echo "unsupported system"
+        exit 1
+        ;;
+    esac
+
+}
+
+create_nested_core_vm(){
+    # determine arch related vars
+    QEMU=$(get_qemu_for_nested_vm)
 
     # create ubuntu-core image
     EXTRA_SNAPS=""
     if [ -d "${PWD}/extra-snaps" ] && [ "$(find "${PWD}/extra-snaps/" -type f -name "*.snap" | wc -l)" -gt 0 ]; then
         EXTRA_SNAPS="--extra-snaps ${PWD}/extra-snaps/*.snap"
     fi
     /snap/bin/ubuntu-image --image-size 3G "$TESTSLIB/assertions/nested-${NESTED_ARCH}.model" --channel "$CORE_CHANNEL" --output ubuntu-core.img "$EXTRA_SNAPS"
-    mkdir -p /tmp/work-dir
-    mv ubuntu-core.img /tmp/work-dir
+    mkdir -p "$WORK_DIR"
+    mv ubuntu-core.img "$WORK_DIR"
```

Comment:
You can pass `-O <directory>` to ubuntu-image and get rid of `mv` here (also, mkdir can go away if you follow my advice regarding mktemp).

Reply:
nice, change done